### ISSUE TITLE
[3.3.1] Incorrect/confusing color scheme legend in layer view

### ISSUE THREAD

**Issue by rickyzhang82** [Ultimaker/Cura#3781]
**Application Version**

3.3.1
**Platform**

Mac/Linux

**Printer**

MP select mini
**Steps to Reproduce**

Open layer view and check the feed rate.

**Actual Results**

As you can see below, ![3.x](https://raw.githubusercontent.com/rickyzhang82/bug-filing/master/cura-3.31-color-legend/3.x.png)

The color legend is very confusing regarding to the bright green vs dark green. You have no clue of feed rate due to the fact that both dark green show in two sides of bright green in the legend.

**Expected results**

I'd expect that bright green is more red shift than the dark green (if you follow the physic rule). As the wave length legend show below, 

![wavelength](https://raw.githubusercontent.com/rickyzhang82/bug-filing/master/cura-3.31-color-legend/wavelength.jpg)

Please change the color legend.

Also, I doubt the color is right. Because the outer wall should print slower than inside. Thus, it should use bright green in the inner wall and dark green in the outer wall in the first diagram.
**Additional Information**

**Comment by rickyzhang82** [comment]
I'd propose the following:
1. I think the color it picked in the diagram needs to be in relative scale, instead of absolute scale for visualization purpose. Dark green vs bright green is harder to interpret than blue vs green. If you only show 5 different speed, you don't need a continuous spectrum and you can tweak it well enough with distinguishable color with discrete color legend. For my eyes, I'd think the inner wall is more bluish than outer wall. Thus, I thought that the print speed is faster in outer wall than inner wall. This is wrong according to the settings. See the diagram below: ![settings](https://raw.githubusercontent.com/rickyzhang82/bug-filing/master/cura-3.31-color-legend/settings.jpg)
1. If picking color is hard to fix, change the color legend first. Use the visible spectrum above is way much better than what we have now.

**Comment by ianpaschal** [comment]
I agree, it could be better. The really quick version in my mind is would be a blue-to-red gradient or something.

**Comment by ChrisTerBeke** [comment]
Actually we specifically avoided red because it usually indicates warning/danger and we didn't want users to think parts of the print were not going to be printed correctly...

The current spectrums were chosen after quite some consideration with several people within Ultimaker.

**Comment by rickyzhang82** [comment]
In layer view, inset is red line. The red line rule needs to be consistent.

![red_line](https://user-images.githubusercontent.com/3475933/40632453-4e09b9ce-62b5-11e8-87bf-5f9d8e938df9.png)

**Comment by ChrisTerBeke** [comment]
Sure, no need to get upset....

**Comment by rickyzhang82** [comment]
Inconsistent rule is a big no no in any logic system.

Back to the fix, ditching the color spectrum should be easiest fix. Just pick several distinguishable color in advance and show the exact print speed with color as a dynamic legend. This alone can solve the problem with minimal code change.

Unless you show a heat map which contains tons of variable difference in the data, the color spectrum is an overkill here that leads to sub par usabilities.

**Comment by ChrisTerBeke** [comment]
> This alone can solve the problem with minimal code change.

In that case, feel free to make a pull request to fix this :)

**Comment by rickyzhang82** [comment]
Well, I'm working on 15.04 legacy branch for [an more important feature](https://github.com/Ultimaker/CuraEngine/pull/782) on my own. 

I live in US and wake up at 4AM to get this feature done asap. I don't have bandwidth to get this UI done.  In addition, I'm a back end developer. Solving UI is never my interest.

:-)

**Comment by ChrisTerBeke** [comment]
Fair enough. We'll see if either the core team or someone from the community picks this up then.

Just a heads up, I don't think the legacy branch will ever be touched again by us, so that PR is probably not going to get merged. If you have ideas for improving the path planning for an up-to-date version of CuraEngine, we'd be happy to review and help out.

**Comment by ianpaschal** [comment]
I can see how "red means under extrusion" might have made people avoid using it for the flow.

I think using 3 primary hues in the spectrum is a mistake, I'd use only two adjacent ones, like blue to green, or yellow to green, etc.

That being said, it's not _broken_ so it won't get any priority from us. It will have to come as a PR from someone.

**Comment by printingotb** [comment]
Hi! Sorry for opening an old discussion.
If we temporarily set travel speed to something as low as the other settings, it becomes much easier to see the difference between different line types.
Perhaps a check-box whether to include travel moves in the preview or not would help?

**Comment by printingotb** [comment]
Sorry, I meant to not include the "travel speed" for calculating the colour map.

**Comment by GregValiant** [comment]
I'm cleaning up these old reports.  I'll go ahead and close this.  If the problem has persisted into 5.13.0 please make out a new report.
